tests/caasp/stack_update.pm

```diff
@@ -36,6 +35,14 @@ sub check_update_changes {
 
     # Kubernetes checks
     assert_script_run "kubectl cluster-info";
+    assert_script_run "kubectl cluster-info > cluster.after_update";
+    if (script_run "diff -Nur cluster.before_update cluster.after_update") {
```

Comment:
Reading your soft_failure message you can (no need to change, your way is also fine - just for info) write this: 
```
if (script_run "kubectl cluster-info | grep '^Dex is running'") {...}
```